plugin.video.cbc/resources/lib/livechannels.py

```diff
@@ -30,34 +31,59 @@ def get_live_channels(self):
             return None
         save_cookies(self.session.cookies)
 
-        items = json.loads(resp.content)[LIST_ELEMENT]
-        return items
+        ret = None
+        for result in json.loads(resp.content)['lineups']['results']:
+            if result['key'] == LIST_ELEMENT:
+                ret = result['items']
+
+        future_to_callsign = {}
+        with futures.ThreadPoolExecutor(max_workers=20) as executor:
+            for i, channel in enumerate(ret):
+                callsign = CBC.get_callsign(channel)
+                future = executor.submit(self.get_channel_metadata, callsign)
+                future_to_callsign[future] = i
+
+        for future in futures.as_completed(future_to_callsign):
+            i = future_to_callsign[future]
+            metadata = future.result()
+            ret[i]['image'] = metadata['Metas']['imageHR']
+        return ret
 
     def get_iptv_channels(self):
         """Get the channels in a IPTV Manager compatible list."""
         cbc = CBC()
         channels = self.get_live_channels()
+        channels = [channel for channel in channels if channel['feedType'].lower() == 'livelinear']
         blocked = self.get_blocked_iptv_channels()
         result = []
         for channel in channels:
             callsign = CBC.get_callsign(channel)
 
             # if the user has omitted this from the list of their channels, don't populate it
-            if callsign in blocked:
+            if f'{callsign}' in blocked:
                 continue
 
             labels = CBC.get_labels(channel)
             image = cbc.get_image(channel)
+
```

Comment:
I am not completely following this? Is the client key something the user needs to add? Or is the add-on circumventing an authorization mechanic?

Reply:
Sorry @basrieter - those notes were added on as I was working through the changes.

Auth is provided when we start the addon (or when their token expires) and the cookies are preserved on disk.

I'll get rid of these so its not so contradictory!